specs/_features/eip7594/das-core.md

```diff
@@ -0,0 +1,222 @@
+# EIP-7594 -- Data Availability Sampling Core
+
+**Notice**: This document is a work-in-progress for researchers and implementers.
+
+## Table of contents
+
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Custom types](#custom-types)
+- [Configuration](#configuration)
+  - [Data size](#data-size)
+  - [Custody setting](#custody-setting)
+  - [Helper functions](#helper-functions)
+    - [`get_custody_lines`](#get_custody_lines)
+    - [`compute_extended_data`](#compute_extended_data)
+    - [`compute_extended_matrix`](#compute_extended_matrix)
+    - [`get_data_column_sidecars`](#get_data_column_sidecars)
+- [Custody](#custody)
+  - [Custody requirement](#custody-requirement)
+  - [Public, deterministic selection](#public-deterministic-selection)
+- [Peer discovery](#peer-discovery)
+- [Extended data](#extended-data)
+- [Column gossip](#column-gossip)
+  - [Parameters](#parameters)
+  - [Reconstruction and cross-seeding](#reconstruction-and-cross-seeding)
+- [Peer sampling](#peer-sampling)
+- [Peer scoring](#peer-scoring)
+- [DAS providers](#das-providers)
+- [A note on fork choice](#a-note-on-fork-choice)
+- [FAQs](#faqs)
+  - [Row (blob) custody](#row-blob-custody)
+  - [Subnet stability](#subnet-stability)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Custom types
+
+We define the following Python custom types for type hinting and readability:
+
+| Name | SSZ equivalent | Description |
+| - | - | - |
+| `DataColumn` | `List[Cell, MAX_BLOBS_PER_BLOCK]` | The data of each column in EIP-7594 |
+| `ExtendedMatrix` | `List[Cell, MAX_BLOBS_PER_BLOCK * NUMBER_OF_COLUMNS]` | The full data of one-dimensional erasure coding extended blobs (in row major format) |
+| `FlatExtendedMatrix` | `List[BLSFieldElement, MAX_BLOBS_PER_BLOCK * FIELD_ELEMENTS_PER_BLOB * NUMBER_OF_COLUMNS]` | The flattened format of `ExtendedMatrix` |
```

Comment:
```suggestion
| `FlatExtendedMatrix` | `List[BLSFieldElement, FIELD_ELEMENTS_PER_CELL * MAX_BLOBS_PER_BLOCK * NUMBER_OF_COLUMNS]` | The flattened format of `ExtendedMatrix` |
```

As discussed in https://github.com/hwwhww/consensus-specs/pull/1#discussion_r1417574161